### DISCUSSION TITLE
Generate REFRESH TOKEN from Google Cloud Console for my Python code

### DISCUSSION THREAD

**Discussion by saeedmv** [fastapi-users/fastapi-users#1323] [Q&A]
Hi, I am trying to generate REFRESH TOKEN from Google Cloud Console for my Python code.  Also, I already have client_id and client_secret.

could you help me generate REFRESH TOKEN from Google Cloud Console or any other way?

My 'Jason code' is as 'client_secret.json' as file name:
`{
  "web": {
    "client_id": "my_client_id",
    "client_secret": "my_client_secre",
    "redirect_uris": ["urn:ietf:wg:oauth:2.0:oob", "http://localhost"]
  }
}`
my 'credentials code' is as 'credentials.json' as file name:
`{
  "token": "my_token",
  "refresh_token": "my_refresh_token",
  "token_uri": "https://oauth2.googleapis.com/token",
  "client_id": "my_client_id",
  "client_secret": "my_client_secret",
  "scopes": [
    "https://www.googleapis.com/auth/drive"
  ],
  "expiry": "2022-01-01T00:00:00Z"
}`
my refresh token code is :
`from google.oauth2.credentials import Credentials
import json

# Load the client secret file
client_secret_file = "client_secret.json"

# Load the credentials file
with open("credentials.json", "r") as f:
    credentials_data = json.load(f)

# Create the credentials object
creds = Credentials.from_authorized_user_info(info=credentials_data)

# Print the refresh token
print("Refresh Token:", creds.refresh_token)`

Is my activities fine?

**Comment by hgalytoby** [comment]
https://fastapi-users.github.io/fastapi-users/12.1/configuration/oauth/

https://github.com/frankie567/httpx-oauth/blob/master/httpx_oauth/clients/google.py

https://github.com/frankie567/httpx-oauth/blob/master/httpx_oauth/oauth2.py#L152


If you have a refresh_token, you can obtain new data by doing the following.
```py
from httpx_oauth.clients.google import GoogleOAuth2

google_oauth_client = GoogleOAuth2('CLIENT_ID', 'CLIENT_SECRET')

data = await google_oauth_client.refresh_token(refresh_token='refresh_token')
print(data)
```

I'm not sure if this is what you're looking for.
